prime-router/src/main/kotlin/azure/SettingsFunctions.kt

```diff
@@ -274,9 +275,13 @@ open class BaseFunction(
         clazz: Class<T>,
         organizationName: String? = null
     ): HttpResponseMessage {
-        authenticateAdmin(request)
-            ?: return HttpUtilities.unauthorizedResponse(request, authenticationFailure)
-
+        val claims = AuthenticatedClaims.authenticate(request)
+        if (claims == null ||
+            !claims.authorized(setOf(PRIME_ADMIN_PATTERN, "$organizationName.*.admin", "$organizationName.*.user"))
```

Comment:
Just noting that `organizationName` is nullable,  I don't think it will have a negative impact since the scope will be `null.*.admin` until we have an org name null.

Could consider a follow up ticket, that splits this function up between fetching all settings vs just fetching an orgs.

Reply:
Will create a new ticket